pkg/flow/internal/controller/loader.go

```diff
@@ -452,7 +452,7 @@ func (l *Loader) populateComponentNodes(g *dag.Graph, componentBlocks []*ast.Blo
 		// Check the graph from the previous call to Load to see we can copy an
 		// existing instance of ComponentNode.
 		if exist := l.graph.GetByID(id); exist != nil {
-			c = exist.(*ComponentNode)
+			c = exist.(*BuiltinComponentNode)
```

Comment:
Let's be a little protective and only cast this to the minimum interface we need:

```suggestion
			c = exist.(ComponentNode)
```